### ISSUE TITLE
testParallelRecombination

### ISSUE THREAD

**Issue by newexo** [garden-of-delete/recombo_public#8]
In order to get test to build, I had to comment out line 73 from `src/tests/recomboCriteriaTests.cpp`
```
    // TODO: RB had to comment out next line to build tests
    // sites=rctc.preConformation.countRecomboSites(18,22,orientation);
```
It seems that at some point, the signature for `countRecomboSites` had changed. More generally, the test `testParallelRecombination` contains significantly many commented lines it is no longer apparent what the intention was for the test.

After commenting out that line from that particular test, the tests built and ran, but with a seg fault. Hence, I further had to comment out line 42 from `src/tests/unitTestMain.cpp`
```
	// TODO: RB had to comment next line because of Segmentation fault 11
	// suite.add_test(testParallelRecombination, "Recombination between unknots");
```

Strictly speaking, we should consider the build to be broken at this point.

**Comment by newexo** [comment]
Oh, my goodness. Oh, no. There are now many, many new parameters to these recombo which do not seem to be tested at all. I am searching for where these are documented. There are also two separate functions that perform recombination depending on whether recombo sites are between different components or in same component.

```
Sequence_type
Recombo_type
Total_para_site
Total_anti_site
Para_site
Anti_site
```

I want to limit the scope of this issue. It seems that testParallelrecombination had been written earlier in history when when the function countRecomboSites only had three parameters. I had to comment out the line 73 to make recomboCriteriaTests.cpp compile. But that was the line which should call countRecomboSites, and documentation explictily says that calling performRecombination without calling countRecomboSites before will have unpredictable results.

Do we know what kind of recombination this test originally performed? Was it parallel recombination?

**Comment by newexo** [comment]
The function tested here is one which may no longer be used.
https://github.com/garden-of-delete/recombo_public/blob/master/src/tests/recomboCriteriaTests.cpp#L74
https://github.com/garden-of-delete/recombo_public/blame/master/src/clkConformationBfacf3.cpp#L1279

I will change the test so that it invokes the version of performRecombination actually used in mmc.

**Comment by newexo** [comment]
Alas. Using the virtual recombination in the same way we are using the old style recombination seems tricky. For the moment, for this one test let us fall back to the old performRecombination function. We should consider creating a follow up task to test virtual recombination.

**Comment by newexo** [comment]
Hey. Looks like this issue should be closed now that we have merged #26 .

Closing.
